src/rapier_body_utils_2d.cpp

```diff
@@ -99,7 +90,9 @@ bool RapierBodyUtils2D::body_motion_recover(
 				Vector2 b(contact.point2.x, contact.point2.y);
 
 				// Compute plane on b towards a.
-				Vector2 n = (a - b).normalized();
+				Vector2 n = Vector2(contact.normal1.x, contact.normal1.y);
+				// Move it outside as to fit the margin
```

Comment:
I'm not comfortable with the idea of changing the algorithm, why did you do that?

Reply:
Is it not the same? Normal1 and a-b normalized is the same value, and its already computed by the function. So i just used that to save computation. Normalized does sqrt inside, so its expensive.